presto-docs/src/main/sphinx/admin/properties.rst

```diff
@@ -416,6 +416,29 @@ Task Properties
     to become overloaded due to excessive resource utilization. This can also be specified on
     a per-query basis using the ``task_writer_count`` session property.
 
+``scale_writers`` and ``writer_min_size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``scale_writers`` config or session property:
+
+    * **Type:**  ``boolean``
+    * **Default value:** ``false``
+
+``writer_min_size`` config or session property:
+
+    * **Type:**  ``data size``
+    * **Default value:** ``32 MB``
+
+    By default, the amount of writer tasks is static. Setting ``scale_writers``
+    to ``true`` enables the ability of Presto to dynamically scale out the
+    number of writer tasks rather than allocating a fixed number of tasks.
+    Additional tasks are added when the average amount of physical data per
+    writer is above a minimum threshold, ``writer_min_size``.
+
+    With writer scaling enabled write queries can take longer to run due to the
+    decreased writer parallelism while the writer count ramps up. Typically
+    however write query performance overall improves.
```

Comment:
Also, it solves the problem where a lot of small files are created due to large number of static write tasks.